doc/troubleshooting.md

```diff
@@ -24,6 +24,68 @@ On Windows the `.cmd` file changes the `CWD` to the location of the script befor
 When you encounter an issue using MotoROS2, please start the debug client script and keep it running in the background while you reproduce the issue.
 Attach the log it produces and a copy of the `PANELBOX.LOG` from the robot's teach pendant to any support tickets you open on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues).
 
+## Network issues
+
+If you are unable to connect the PC running the micro-ROS agent application and the controller, the network configuration may be the issue. Network problems can sometimes be the root of seemingly unrelated problems, such as rcl/rclc errors.
+
+To troubleshoot network issues, first ensure that the robot controller's IP address is `MANUAL SETTING`. It is important that it is not using DHCP. [See here](../README.md#checking-network-configuration).
+
+If that is not the problem, check that the `agent_ip_address` matches the IP address of the PC. Also make sure that the robot controller IP address is on the same subnet as the PC.
+
+If there is a mismatch somewhere, choose one of the following options:
+
+1. Modify the `agent_ip_address` key in the `motoros2_config.yaml` file and specify an IP address that is on the robot's subnet. Then [propagate the changes to the Yaskawa controller](../README.md#updating-the-configuration). You will need to ensure that the PC running the micro-ROS agent application uses this static IP address on the network port connected to the robot controller. You will also need to make sure tha the subnet mask matches that of the controller.
+2. Modify the robot controller's IP and subnet mask so it is on the subnet of the PC running the micro-ROS agent.
+3. Modify the robot controller's network settings to add a gateway which can reach the Agent's IP address.
+
+Refer to the relevant Yaskawa Motoman documentation for information on how to change the controller's network configuration.
+
+Test if the two are connected by opening a terminal on the PC and attempting to ping the robot controller's IP. For example
+
+```shell
+ping 192.168.1.31
+```
+
+If you receive a near-immediate response upon attempting to ping the controller, then the potential network issues mentioned above are resolved. Try to connect again via the micro-ROS agent. If the micro-ROS agent still cannot connect to the controller, there may be a firewall configuration issue.
+
+If you do not receive a near-immediate after attempting to ping the controller, there is still some issue. You may have made a mistake previously, or there could be a firewall issue, or there could be a network hardware issue.
```

Comment:
Maybe suggest plugging PC directly into the robot (bypassing any additional hardware, such as switches). Also suggest trying a different cable.